lib/gui/components/safe-webview/safe-webview.js

```diff
@@ -0,0 +1,231 @@
+
+          // Hide/show the webview per the documented way
+          // https://electron.atom.io/docs/api/webview-tag/#css-styling-notes
+          flex: this.state.shouldLoad ? null : '0 1',
+          width: this.state.shouldLoad ? null : '0',
+          height: this.state.shouldLoad ? null : '0'
+        }
+      }, []);
+
+    }
+
+    // We have to return null explicitly, undefined is an error in React.
+    return null;
+  }
+
+  /**
+   * @summary Add the Webview events if there is an element
+   */
+  componentDidMount() {
+
+    // There is no element to add events to if 'shouldLoad' is false.
+    if (this.state.shouldLoad) {
+
+      // Events React is unaware of have to be handled manually
+      this.refs.webview.addEventListener('did-start-loading', this.didStartLoading);
+      this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
+      this.refs.webview.addEventListener('did-stop-loading', this.didStopLoading);
+      this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
+      this.refs.webview.addEventListener('will-navigate', this.willNavigate);
+      this.refs.webview.addEventListener('ipc-message', this.ipcMessage);
+    }
+  }
+
+  /**
+   * @summary Remove the Webview events if there is an element
+   */
+  componentWillUnmount() {
+
+    // There is no element to remove events from if 'shouldLoad' is false.
+    if (this.state.shouldLoad) {
+
+      // Events React is unaware of have to be handled manually
+      this.refs.webview.removeEventListener('did-start-loading', this.didStartLoading);
+      this.refs.webview.removeEventListener('did-fail-load', this.didFailLoad);
+      this.refs.webview.removeEventListener('did-stop-loading', this.didStopLoading);
+      this.refs.webview.removeEventListener('did-get-response-details', this.didGetResponseDetails);
+      this.refs.webview.removeEventListener('will-navigate', this.willNavigate);
+      this.refs.webview.removeEventListener('ipc-message', this.ipcMessage);
+    }
+  }
+
+  /**
+   * @summary Start the loading timeout timer
+   */
+  didStartLoading() {
+    if (this.props.timeout) {
+      this.loadingTimeoutId = setTimeout(() => {
+        this.setState({
+          shouldLoad: false
+        });
+
+      // Milliseconds to wait before cancelling the page-load.
+      }, this.props.timeout);
+    }
+  }
+
+  /**
+   * @summary Set the element state to hidden
+   */
+  didFailLoad() {
+    this.setState({
+      shouldLoad: false
+    });
+  }
+
+  /**
+   * @summary Remove the timeout
+   */
+  didStopLoading() {
+
+    // We have loaded, remove the queued timeout action.
+    clearTimeout(this.loadingTimeoutId);
+  }
+
+  /**
+   * @summary Set the element state depending on the HTTP response code
+   * @param {Event} event - Event object
+   */
+  didGetResponseDetails(event) {
+
+    const {
+      httpResponseCode
+    } = event;
+    const HTTP_OK = 200;
+
+    if (httpResponseCode !== HTTP_OK) {
```

Comment:
Can we do `event.httpResponseCode`?

```js
const {
   httpResponseCode
} = event;
```

Looks very weird and this is the only place where this variable is used.